paper/paper.md

```diff
@@ -130,47 +129,49 @@ end
+After completing the necessary steps, one can utilize a solver from JSOSolvers to minimize the loss of LeNetNLPModel. These solvers have been primarily designed for deterministic optimization. In the case of FluxNLPModel.jl (and KnetNLPModels.jl), the loss function is managed to ensure its application to sampled data. However, it is essential to modify the training minibatch between iterations. This can be accomplished by leveraging the callback mechanism incorporated in JSOSolvers. This mechanism executes a pre-defined function, known as a callback, at the conclusion of each iteration. For more comprehensive information, please consult the JSOSolvers [@jso] documentation.
 
 In the following code snippet, we demonstrate the execution of the R2 solver with a `callback` that changes the training minibatch at each iteration:
 ```julia
+using JSOSolvers
+
 max_time = 300. # run at most 5min
-callback = (nlpmodel, solver, stats) -> FluxNLPModels.minibatch_next_train!(nlpmodel)
+callback = (LeNetNLPModel, solver, stats) -> FluxNLPModels.minibatch_next_train!(LeNetNLPModel)
 
-solver_stats = JSOSolvers.R2(LeNetNLPModel; callback, max_time)
+solver_stats = R2(LeNetNLPModel; callback, max_time)
 
 ## Report on test data
-test_accuracy = FluxNLPModels.accuracy(nlp)
+test_accuracy = FluxNLPModels.accuracy(LeNetNLPModel)
 ```
 
-Another choice to train `LeNetNLPModel` is the LBFGS linesearch solver of JSOSolvers.jl:
+Another choice to train `LeNetNLPModel` is the LBFGS linesearch solver:
 ```julia
 solver_stats = lbfgs(LeNetNLPModel; callback, max_time)
-
 test_accuracy = FluxNLPModels.accuracy(LeNetNLPModel)
 ```
 
-To enhance the `LeNetNLPModel`, an `LSR1` (or `LBFGS`) approximation of the Hessian can be employed and fed into the `trunk` solver. The trunk solver utilizes a quadratic trust-region method with a backtracking line search. To integrate the LSR1 approximation and trunk into the training process, the code can be modified as outlined below:
+To `LeNetNLPModel`, an `LSR1` approximation of the Hessian can be employed and fed into the `trunk` solver. The trunk solver utilizes a trust-region method with a backtracking line search. To integrate the LSR1 approximation and trunk into the training process, the code can be modified as:
 
 ```julia
-using NLPModelsModifiers # define also LBFGSModel
+using NLPModelsModifiers # define also LSR1Model
 
 lsr1_LeNet = NLPModelsModifiers.LSR1Model(LeNetNLPModel)
-
 callback_lsr1 = 
-  (sr1_nlpmodel, solver, stats) -> FluxNLPModels.minibatch_next_train!(sr1_nlpmodel.model)
+  (lsr1_LeNet, solver, stats) -> FluxNLPModels.minibatch_next_train!(lsr1_LeNet)
```

Comment:
```suggestion
  (lsr1_LeNet, solver, stats) -> FluxNLPModels.minibatch_next_train!(
                                                 lsr1_LeNet.model
                                               )
```

Reply:
After some thoughts, there was no mistake with `sr1_nlpmodel`, it is just a variable name in an anonymous function definition.